src/openscm_runner/adapters/fair2/fair2.py

```diff
@@ -0,0 +1,316 @@
+                        'Emissions|Montreal Gases|CH2Cl2',
+                        'Emissions|Montreal Gases|CHCl3',
+                    ]
+
+                    continue
+
+                if in_unit != fair_unit:
+                    ssp_df_hist = ssp_df_hist.convert_unit(fair_unit, variable=variable, context=context)
+
+            self._values_fair_history = ssp_df_hist
+
+        self._loaded_fair_history = True
+
+        return self._values_fair_history
+
+
+
+# TODO: lazy load this and only load once
+ssp245_world_emms_holder = SSP245WorldEmms()
+
+
+EMISSIONS_SPECIES_UNITS_CONTEXT = pd.DataFrame((
+        ('|CO2|MAGICC Fossil and Industrial', 'GtC / yr', None),
+        ('|CO2|MAGICC AFOLU', 'GtC / yr', None),
+        ('|CH4', 'MtCH4 / yr', None),
+        ('|N2O', 'MtN / yr', None),
+        ('|Sulfur', 'MtS / yr', None),
+        ('|CO', 'MtCO / yr', None),
+        ('|VOC', 'MtNMVOC / yr', None),
+        ('|NOx', 'MtN / yr', "NOx_conversions"),
+        ('|BC', 'MtBC / yr', None),
+        ('|OC', 'MtOC / yr', None),
+        ('|NH3', 'MtN / yr', None),
+        ('|CF4', 'ktCF4 / yr', None),
+        ('|C2F6', 'ktC2F6 / yr', None),
+        ('|C6F14', 'ktC6F14 / yr', None),
+        ('|HFC23', 'ktHFC23 / yr', None),
+        ('|HFC32', 'ktHFC32 / yr', None),
+        ('|HFC4310mee', 'ktHFC4310mee / yr', None),
+        ('|HFC125', 'ktHFC125 / yr', None),
+        ('|HFC134a', 'ktHFC134a / yr', None),
+        ('|HFC143a', 'ktHFC143a / yr', None),
+        ('|HFC227ea', 'ktHFC227ea / yr', None),
+        ('|HFC245fa', 'ktHFC245fa / yr', None),
+        ('|SF6', 'ktSF6 / yr', None),
+        ('|CFC11', 'ktCFC11 / yr', None),
+        ('|CFC12', 'ktCFC12 / yr', None),
+        ('|CFC113', 'ktCFC113 / yr', None),
+        ('|CFC114', 'ktCFC114 / yr', None),
+        ('|CFC115', 'ktCFC115 / yr', None),
+        ('|CCl4', 'ktCCl4 / yr', None),
+        ('|CH3CCl3', 'ktCH3CCl3 / yr', None),
+        ('|HCFC22', 'ktHCFC22 / yr', None),
+        ('|HCFC141b', 'ktHCFC141b / yr', None),
+        ('|HCFC142b', 'ktHCFC142b / yr', None),
+        ('|Halon1211', 'ktHalon1211 / yr', None),
+        ('|Halon1202', 'ktHalon1202 / yr', None),
+        ('|Halon1301', 'ktHalon1301 / yr', None),
+        ('|Halon2402', 'ktHalon2402 / yr', None),
+        ('|CH3Br', 'ktCH3Br / yr', None),
+        ('|CH3Cl', 'ktCH3Cl / yr', None),
+    ),
+    columns=["species", "in_unit", "context"],
+)
+
+def _get_fair_col_unit_context(variable):
+    row = EMISSIONS_SPECIES_UNITS_CONTEXT["species"].apply(lambda x: variable.endswith(x))
+
+    in_unit = EMISSIONS_SPECIES_UNITS_CONTEXT[row]["in_unit"]
+
+    assert in_unit.shape[0] == 1, in_unit
+
+    fair_col = int(row[row].index.values) + 1  # first col is time
+    in_unit = in_unit.iloc[0]
+    context = EMISSIONS_SPECIES_UNITS_CONTEXT[row]["context"].iloc[0]
+
+    return fair_col, in_unit, context
+
+
+def _scmdf_to_emissions_df(scmdf, include_cfcs=True, startyear=1765, endyear=2100):
```

Comment:
It would be great to a) test this more thoroughly and b) write it in a much more flexible way than the current implementation